tests/backup/backup_helper.go

```diff
@@ -1181,13 +1247,128 @@ func backupSuccessCheck(backupName string, orgID string, retryDuration int, retr
 
 	_, err := task.DoRetryWithTimeout(backupSuccessCheck, time.Duration(retryDuration)*time.Minute, time.Duration(retryInterval)*time.Second)
 	if err != nil {
-		return false, err
+		return false, nil, err
 	}
-	return true, nil
+	backupInspectRequest := &api.BackupInspectRequest{
+		Name:  backupName,
+		Uid:   bkpUid,
+		OrgId: orgID,
+	}
+	resp, err := backupDriver.InspectBackup(ctx, backupInspectRequest)
+	if err != nil {
+		return false, nil, err
+	}
+	backupStatus := (resp.GetBackup().GetStatus().Status == api.BackupInfo_StatusInfo_Success)
+	log.Infof("Backup [%s] created successfully", backupName)
+	return backupStatus, resp, nil
+}
+
+func ValidateBackup(backupInspectResponse *api.BackupInspectResponse, clusterAppsContexts []*scheduler.Context) (*BackupRestoreContext, error) {
+	bkpObjs := backupInspectResponse.Backup.Resources
+	if backupSpecContext, err := GetBackupSpecObjectsContexts(bkpObjs, clusterAppsContexts); err != nil {
+		return nil, fmt.Errorf("GetBackupSpecObjects Err: %v", err)
+	} else {
+		return &BackupRestoreContext{
+			schedCtxs: backupSpecContext,
+		}, nil
+	}
+}
+
+// Filters the source namespaces' contexts to only include backup objects
+// MAKE SURE to include Contexts of ALL namespaces which contain the backup objects
+func GetBackupSpecObjectsContexts(resourceInfos []*api.ResourceInfo, clusterAppsContexts []*scheduler.Context) ([]*scheduler.Context, error) {
+	log.InfoD("Validation: Getting the backup objects (specs) from the Context")
+
+	var backupAppContexts []*scheduler.Context = make([]*scheduler.Context, 0)
+	var allBkpObjs []*api.ResourceInfo = make([]*api.ResourceInfo, 0)
+
+	//for each clusterAppsContext (namespace), we return the corresponding BackupSpecObject
+	for _, clusterAppsContext := range clusterAppsContexts {
+		var specObjects []interface{} = make([]interface{}, 0)
+		// the Backup objects which were found in this namespace
+		var resInfoObjs []*api.ResourceInfo = make([]*api.ResourceInfo, 0)
+
+		for _, resource := range resourceInfos {
+			if resource.Namespace == clusterAppsContext.ScheduleOptions.Namespace {
+				resInfoObjs = append(resInfoObjs, resource)
+			} else if resource.Namespace == "" {
+				// IF it is therein this context, we'll work with it
+				resInfoObjs = append(resInfoObjs, resource)
+			}
+		}
+
+	specloop:
+		for _, spec := range clusterAppsContext.App.SpecList {
+			name, kind, ns, err := GetSpecNameKindNamepace(spec)
+			if err != nil {
+				log.Error(err)
+				continue specloop
+			}
+			if name != "" /*&& kind != ""*/ /*kind is empty for certain objects :(*/ {
```

Comment:
As far as I can say, there is a bug that originates elsewhere in the code where `kind` field is empty for some of the resources (not all). 

If that bug hadn't existed, then I would also compare kind _along with name_ as the same name might be given to multiple resources of different kinds (which is actually the case in `elastic-search-crd` and this will result in incorrect logic. So, I've left it in there so there as a reminder that it can fail (so anyone who looks at it in the future immediately knows the BUG).

Reply:
I have verified that this is a known issue. I'll file an issue and leave it as a commented out part (of that's alright). It should be uncommented in the future, when there's bug fix for the missing `kind` value.